subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -1367,3 +1367,46 @@ Gradle 5.3 introduced the interface `TransformAction` for implementing artifact
 Using the registration method link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyHandler.html#org.gradle.api.artifacts.dsl.DependencyHandler:registerTransform(org.gradle.api.Action)[DependencyHandler.registerTransform(Action)] for `ArtifactTransform` has been deprecated.
 Migrate your artifact transform to use `TransformAction` and use link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyHandler.html#org.gradle.api.artifacts.dsl.DependencyHandler:registerTransform(java.lang.Class,%20org.gradle.api.Action)[DependencyHandler.registerTransform(Class, Action)] instead.
 See the <<artifact_transforms#sec:abm_artifact_transforms,user manual>> for more information on implementing `TransformAction`.
+
+[[gmm_modification_after_publication_populated]]
+==== Modifying GMM after a publication has been populated
+
+Altering the link:publishing_gradle_module_metadata.html[Gradle Module Metadata] (e.g. changing a component configuration variants) after a Maven or Ivy publication have been populated from their components, is now deprecated and scheduled to be disallowed in Gradle 9.0.
+
+Eager "realization" of the publication can happen if the following methods are called:
```

Comment:
```suggestion
Eager population of the publication can happen if the following methods are called:
```
I'm not sure what you mean by "realization" in this sentence. Should we use "population" instead, since you use it in the previous sentence? Otherwise, we should consider introducing the concept of "realization" and relating it to the existing concept of population.